src/domain/service/note.ts

```diff
@@ -71,4 +61,14 @@ export default class NoteService {
       content: addedNote.content,
     };
   }
+
+  /**
+   * Gets note by id
+   *
+   * @param options - get note by id options
+   * @returns { Promise<Note | null> } note
+   */
+  public async getNoteById({ id }: GetNoteByIdOptions): Promise<Note | null> {
```

Comment:
seems redundant, lets just use `getNoteById(id: string)` here.

Or `getNoteById(id: NoteId)` where NoteId is a Nominal type

Reply:
done